src/java_bytecode/java_types.h

```diff
@@ -432,6 +443,33 @@ inline const optionalt<size_t> java_generics_get_index_for_subtype(
   return std::distance(gen_types.cbegin(), iter);
 }
 
+/// Get the index in the subtypes array for a given component.
+/// \param t The type we search for the subtypes in.
+/// \param identifier The string identifier of the type of the component.
+/// \return Optional with the size if the identifier was found.
+inline const optionalt<size_t> java_implicit_generics_get_index_for_subtype(
+  const java_implicitly_generic_class_typet &t,
+  const irep_idt &identifier)
+{
+  const std::vector<java_generic_parametert> &gen_types=
```

Comment:
It would be good to make the common code between this method and `java_generics_get_index_for_subtype` shared somehow.

Reply:
Perhaps by making it take a `std::vector<java_generic_parametert>` as input rather than either of the types